extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -132,9 +132,9 @@ public static void startServerAfterFailedStart() {
     }
 
     public void finalizeRouter(BeanContainer container, Handler<HttpServerRequest> defaultRouteHandler,
-            List<Handler<RoutingContext>> filters, LaunchMode launchMode, ShutdownContext shutdown) {
+            List<Handler<RoutingContext>> orderedFilters, LaunchMode launchMode, ShutdownContext shutdown) {
 
-        for (Handler<RoutingContext> filter : filters) {
+        for (Handler<RoutingContext> filter : orderedFilters) {
```

Comment:
But aren’t you still adding them in reverse order below?

Reply:
does not need to be reversed, the sorting is done in the processor.

Reply:
But by doing:
```
router.route().order(-1).handler(filter);
```
aren't you inserting them in reverse order? I mean the last one will be first, won't it? Or did I miss something?

Reply:
If security has ordinal 100, and cors ordinal set to 20, it will add:

security (because of 200 > 20), and then cors. 

why do you need to reverse them?

Reply:
so on a request, the security filter will be call first, and then the cors filter and then the user route.

Reply:
Thanks for the comment, it makes things far clearer.